### DISCUSSION TITLE
Understanding $watch

### DISCUSSION THREAD

**Discussion by Chandler-Zhu** [alpinejs/alpine#1539] [1. Help] [answered]
This is the `$watch` example in the official v3 doc
![image](https://user-images.githubusercontent.com/61914365/122444103-38ba0780-cf6e-11eb-86a2-bfc2d3be73ba.png)

My question is does `$watch` always comes with `x-init` ? To me `x-init` only gets called _once_ in the entire lifespan of a component, but `$watch` is watching for changes _all the time_ , this confuses me a lot. If the data being watched changes, does the re-render go through an `alpine:initializing` phase as well ?

**Comment by KevinBatdorf** [comment, marked as answer]
When you use `$watch` you're telling Alpine to monitor a property long term (`open` in your example) and fire a callback (`console.log()`) when that property changes.  `x-init` is the perfect place for that since you only want it to register the watcher one time.

Maybe  a good way to think about it is you're "registering" a long-running watcher?
